### PR TITLE
Add check for archived dataset

### PR DESCRIPTION
**Pull request by vpnu**
# Changes

- Check if the scan reports in the active scan reports list belong to datasets that are not archived
- If dataset is active then keep scan report in active scan reports list

# Migrations

# Screenshots

# Checks

**Important:** please complete these **before** merging.
- [ ] Run migrations, if any.
- [x] Update `changelog.md`, including migration instructions if any.
- [ ] Run unit tests.

### CODE REVIEW

File: ProcessQueue/__init__.py [hunk 1]
```diff
@@ -694,15 +700,19 @@ def reuse_existing_value_concepts(new_values_map, content_type, api_url, headers
...
+        headers=headers,
+    )
     # get active scanreports and map them to fields. Remove any fields in archived
     # reports or not marked as 'Mapping Complete'
-    active_srs = [
-        str(item["id"])
-        for item in json.loads(get_scan_reports.content.decode("utf-8"))
-        if item["hidden"] is False and item["status"] == "COMPLET"
-    ]
-    # active reports is list of report ids that belong to an active dataset, are not archived, and have the status
```
Comment (spco) [inline ProcessQueue/__init__.py:704]: Please re-add this comment

File: ProcessQueue/__init__.py [hunk 2]
```diff
@@ -407,13 +407,19 @@ def reuse_existing_field_concepts(new_fields_map, content_type, api_url, headers
         url=f"{api_url}scanreports/",
         headers=headers,
     )
+    get_datasets = requests.get(
+        url=f"{api_url}datasets/",
+        headers=headers,
+    )
     # get active scanreports and map them to fields. Remove any fields in archived
```
Comment (spco) [inline ProcessQueue/__init__.py:414]: Please update this comment (to match the one in the other function)

**Review by spco — COMMENTED** [review]
Good to go after these are addressed

File: ProcessQueue/__init__.py [hunk 3]
```diff
@@ -407,13 +407,19 @@ def reuse_existing_field_concepts(new_fields_map, content_type, api_url, headers
...
     # reports or not marked as 'Mapping Complete'
-    active_srs = [
-        str(item["id"])
-        for item in json.loads(get_scan_reports.content.decode("utf-8"))
-        if item["hidden"] is False and item["status"] == "COMPLET"
-    ]
+    active_srs = []
+    for item in json.loads(get_scan_reports.content.decode("utf-8")):
```
Comment (dcl10) [inline ProcessQueue/__init__.py:417]: These `json.loads` statements could be simplified to `get_scan_reports.json()`

File: ProcessQueue/__init__.py [hunk 4]
```diff
@@ -407,13 +407,19 @@ def reuse_existing_field_concepts(new_fields_map, content_type, api_url, headers
...
+        if item["hidden"] is False and item["status"] == "COMPLET":
+            for ds in json.loads(get_datasets.content.decode("utf-8")):
```
Comment (dcl10) [inline ProcessQueue/__init__.py:419]: See above comment

File: ProcessQueue/__init__.py [hunk 5]
```diff
@@ -694,15 +700,19 @@ def reuse_existing_value_concepts(new_values_map, content_type, api_url, headers
...
-        str(item["id"])
-        for item in json.loads(get_scan_reports.content.decode("utf-8"))
-        if item["hidden"] is False and item["status"] == "COMPLET"
-    ]
-    # active reports is list of report ids that belong to an active dataset, are not archived, and have the status
-    # 'Mapping Complete'
+    active_srs = []
+    for item in json.loads(get_scan_reports.content.decode("utf-8")):
+        if item["hidden"] is False and item["status"] == "COMPLET":
+            for ds in json.loads(get_datasets.content.decode("utf-8")):
```
Comment (dcl10) [inline ProcessQueue/__init__.py:712]: See above comment

**Pull request merged**